v-next/hardhat-utils/src/synchronization.ts

```diff
@@ -123,4 +136,28 @@ export class MultiProcessMutex {
       throw new FileSystemAccessError(e.message, e);
     }
   }
+
+  async #isMutexProcessOwnerDead(): Promise<boolean> {
+    let mutexPid: string;
+
+    try {
+      // If the file doesn't exist, it means the owning process deleted it
+      mutexPid = await readUtf8File(this.#mutexFilePath);
+    } catch (_e) {
+      return false;
+    }
+
+    try {
+      process.kill(parseInt(mutexPid, 10), 0);
+    } catch (e) {
+      ensureError<Error & { code: string }>(e);
```

Comment:
nit: you can use `ensureNodeErrnoExceptionError` instead of passing `Error & { code: string }` as a type parameter.